crates/ink/ir/src/ir/utils.rs

```diff
@@ -174,13 +176,12 @@ pub fn extract_cfg_attributes(
 ) -> Vec<proc_macro2::TokenStream> {
     attrs
         .iter()
-        .filter(|a| a.path.is_ident(super::CFG_IDENT))
+        .filter(|a| a.path().is_ident(super::CFG_IDENT))
         .map(|a| {
-            a.tokens
-                .clone()
-                .into_iter()
-                .map(|token| quote::quote_spanned!(span=> #[cfg #token]))
-                .collect()
+            let expr = a
+                .parse_args::<Expr>()
+                .expect("Encountered cfg attribute in wrong format.");
+            quote::quote_spanned!(span=> #[cfg( #expr )])
```

Comment:
Why parse it? Simply `quote::quote_spanned!(span=> #a)` may work.

If it does we could even take it a step further and make this fn return `Vec<syn::Attribute>` directly.

Reply:
The first approach would work better because we want to return an attribute in spanned context.